src/gui/menu_list.cpp

```diff
@@ -0,0 +1,24 @@
+#include "gui/menu_item.hpp"
+#include "gui/menu_list.hpp"
+#include "gui/menu_manager.hpp"
+
+ListMenu::ListMenu(std::vector<std::string>* items_, int* selected_) : 
+  items(items_), 
+  selected(selected_)
```

Comment:
Why is the list of items saved in the object? The list might not exist anymore if the list was created dynamically, like:
```cpp
...
{
  std::vector<std::string> items = {};
  int selected;
  MenuManager::instance().push_menu(std::make_unique<ListMenu<(&items, &selected));
}
// Both `items` and `selected` are freed, the menu now holds dangling pointers
...
```

Reply:
I didn't think of that at all, yeah I'll remove that part since it's not needed at all.